File: Alkitab/src/main/java/yuku/alkitab/base/widget/VerseInlineLinkSpan.java
package yuku.alkitab.base.widget;

import android.text.TextPaint;
import android.text.style.ClickableSpan;
import android.view.View;
import androidx.annotation.NonNull;

public abstract class VerseInlineLinkSpan extends ClickableSpan {
	public interface Factory {
		VerseInlineLinkSpan create(final Type type, final int arif);
	}

	private final Type type;
	private final int arif;

	public enum Type {
		footnote,
		xref,
	}

	public VerseInlineLinkSpan(final Type type, final int arif) {
		this.type = type;
		this.arif = arif;
	}

	@Override
	public final void onClick(@NonNull final View widget) {
		onClick(type, arif);
	}

	public void onClick(final Type type, final int arif) {
	}

	@Override
	public void updateDrawState(@NonNull final TextPaint ds) {
		// don't call super to prevent link underline and link coloring
		// NOP
	}
}

File: Alkitab/src/main/java/yuku/alkitab/base/widget/VerseInlineLinkSpan.kt
package yuku.alkitab.base.widget

import android.text.TextPaint
import android.text.style.ClickableSpan
import android.view.View

abstract class VerseInlineLinkSpan(private val type: Type, private val arif: Int) : ClickableSpan() {
    fun interface Factory {
        fun create(type: Type, arif: Int): VerseInlineLinkSpan
    }

    enum class Type {
        footnote, xref
    }

    override fun onClick(widget: View) {
        onClick(type, arif)
    }

    open fun onClick(type: Type, arif: Int) {}

    override fun updateDrawState(ds: TextPaint) {
        // don't call super to prevent link underline and link coloring
        // NOP
    }
}